platform/core/src/services/UICustomizationService/UICustomizationService.ts

```diff
@@ -0,0 +1,273 @@
+import merge from 'lodash.merge';
+import { PubSubService } from '../_shared/pubSubServiceInterface';
+import CustomizationTypeInfo from './CustomizationTypeInfo';
+import { UICustomization, UIConfiguration, ExtensionManager } from './types';
+import { ComponentType } from 'react';
+
+const EVENTS = {
+  MODE_CUSTOMIZATION_MODIFIED: 'event::UICustomizationService:modeModified',
+  GLOBAL_CUSTOMIZATION_MODIFIED: 'event::UICustomizationService:globalModified',
+};
+
+type Obj = Record<string, unknown>;
+
+type ComponentReturn = {
+  component: ComponentType;
+  props?: Obj;
+};
+
+type NestedStrings = string[] | NestedStrings[];
+
+const flattenNestedStrings = (
+  strs: NestedStrings | string,
+  ret?: Record<string, string>
+): Record<string, string> => {
+  if (!ret) ret = {};
+  if (!strs) return ret;
+  if (Array.isArray(strs)) {
+    for (const val of strs) {
+      flattenNestedStrings(val, ret);
+    }
+  } else {
+    ret[strs] = strs;
+  }
+  return ret;
+};
+
+/**
+ * The UICustomizationService allows for retrieving of custom components
+ * and configuration for mode and global values.
+ * The intent of the items is to provide a react component.  This can be
+ * done by straight out providing an entire react component or else can be
+ * done by configuring a react component, or configuring a part of a react
+ * component.  These are intended to be fairly indistinguishable in use of
+ * it, although the internals of how that is implemented may need to know
+ * about the customization service.
+ *
+ * A customization value can be:
+ *   1. React function, taking (React, props) and returning a rendered component
+ *      For example, createLogoComponentFn renders a component logo for display
+ *   2. Custom UI component configuration, as defined by the component which uses it.
+ *      For example, context menus define a complex structure allowing site-determined
+ *      context menus to be set.
+ *   3. A string name, being the extension id for retrieving one of the above.
+ *
+ * The default values for the extension come from the app_config value 'whiteLabeling',
```

Comment:
Is this valid anymore? I thought you changed them so that it does not live in `whiteLabeling`